helpers/compile/plugins/cleanupPlugin.ts

```diff
@@ -0,0 +1,32 @@
+import type * as esbuild from 'esbuild'
+import { promisify } from 'util'
+import fs from 'fs'
+
+const unlink = promisify(fs.unlink)
+
+/**
+ * Removes the esm output files after cjs compile.
+ * @param build the esbuild build object
+ * @returns
+ */
+function removeIntermediaryEsmFiles(build: esbuild.PluginBuild) {
+  const files = Object.values(build.initialOptions.entryPoints ?? {})
+  const rms = files.map((file) => file.endsWith('.mjs') && unlink(file))
```

Comment:
```suggestion
  const rms = files
  	.filter((file) => file.endsWith('.mjs'))
  	.map((file) => unlink(file))
```

That's probably fewer promises created like this